Implementation of the purchasing function by payjp

MasatakaOba opened the pull request:
## WHAT
Add Implementation of the purchasing function.

The function details are below. 
1. Credit card registration/delation function.
2. Buying items function

These functions is in cooperation with Payjp database.
I have already conform that  cooperation is normally performed.

notice: Item database has not been created yet. So shortly after created, I will link database to this function.

## WHY
To be able to purchase items.

## Page transition images(GIF)

### Credit card registration/delation
![f25b48e702231a9af6b1f253f080f25f](https://user-images.githubusercontent.com/27143270/56089106-c69a9380-5ec8-11e9-9282-846b448df223.gif)

### Purchasing an item page image
![1f489e069dc11c86a3b67a77bd7b74ec](https://user-images.githubusercontent.com/27143270/56089109-d023fb80-5ec8-11e9-991d-1abd0eab9f31.gif)

Commit:
make function of credit card registration

Final purpose is purchasing items, so I create this function.
To make function of credit card registration, Payjp API is used.

1. According to Payjp website, wright code of javascript pay.jp and add script(src: "https://js.pay.jp/") to views/layouts/application.html.haml
2.Rewright card registration view pages to adjust for payjp API
3.Wright route to make transition.

Commit:
Add some file to register payjp costomer and card ID to database

make the database for card to recall credit card information

Commit:
make the function of purchasing items

According to Payjp website, create and rewrite files for buying items.
1. Create controller to connect payjpAPI to item fee
2. Rewrite views/purchase/index.html to adjust API
3. Create file to conform completion of the transaction

notice: item database has not beet created, so some code will weright to connect with databese valiables in the near future.

yuki-uchida commented on app/controllers/purchase_controller.rb:
```diff
@@ -1,2 +1,30 @@
+    if card.blank?
+      redirect_to controller: "card", action: "new"
+    else
+      require 'payjp'
```
require 'payjp'は2行目にまとめても良いのでは？

MasatakaOba replied:
クラスの呼び出しごとにrequire 'payjp'を呼び出してしまって動作が重くなってしまうと考え分けておりました。
こちら記述を2行目に統一しました。

Commit:
Omit description of "require 'payjp'" and gather one position

Move description of "require 'payjp'" to header of files and delete others within methots.

yuki-uchida commented:
LGTMです

MasatakaOba commented:
> LGTMです

ありがとうございます！
